kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/CachingKms.java

```diff
@@ -89,6 +100,17 @@ public Serde<E> edekSerde() {
     @NonNull
     @Override
     public CompletionStage<K> resolveAlias(@NonNull String alias) {
-        return resolveAliasCache.get(alias);
+        Long notFoundSince = notFoundAliasCache.getIfPresent(alias);
+        if (notFoundSince != null) {
+            return CompletableFuture.failedFuture(new UnknownAliasException("alias " + alias + " not found, cached since " + notFoundSince));
```

Comment:
I'm not sure including/logging the System.currentTimeMillis of the initial entry is overly helpful. Maybe we should convert it to a date or even add it as a field to the `UnknownAliasException` for properly formatting later.

Reply:
Have refactored it to make the cached value the failed future and got rid of the idea of exposing the cache time. There is a debug log available when we decide to cache and the UnknownAliasException message contains `(cached result)`.